internal/util/diagtree/diagtree.go

```diff
@@ -180,20 +180,23 @@ func (m *Node) severity() string {
 // The severity of a node.
 //
 // Nodes with their own (non-None) severity are always displayed on their own level.
-type Severity string
+type Severity struct{ s string }
```

Comment:
What do you think about having an exported struct field called `Level`; then we wouldn't need the `String()` getter below? Personally I'd have an easier time reading that.